src/triggers.js

```diff
@@ -837,13 +846,17 @@ export function maybeRunTrigger(
         resolve(object);
       },
       error => {
-        logTriggerErrorBeforeHook(
-          triggerType,
-          parseObject.className,
-          parseObject.toJSON(),
-          auth,
-          error
-        );
+        try {
+          logTriggerErrorBeforeHook(
+            triggerType,
+            parseObject.className,
+            parseObject.toJSON(),
+            auth,
+            error
+          );
+        } catch (error2) {
+          // https://github.com/parse-community/parse-server/issues/7192
```

Comment:
We avoid empty catch blocks, can you refactor this?

Reply:
How should the code look like?

Reply:
@mtrezza Log the error again without parseObject.toJSON throwing

Reply:
This is bad code quality and incorrect for the reasons I outlined above. It is catching any error of `logTriggerErrorBeforeHook` but you expect only `toJson` to throw. Please refactor to only `try` the `toJson`.

Also, it seems to have to catch twice if `toJson` throws, once in `_getSaveJson` and once in  `logTriggerErrorBeforeHook`. This should be simplified.

Reply:
@mtrezza What you said make sense. I'll make the change

Reply:
I've made the first change to only try catch on the toJSON.

I'm not sure how the 2nd one should be simplified. So I figured out a bug that make toJSON throws. It is used in 2 places, so we try catch it in two places. This seems OK to me.